core/vm/contracts.go

```diff
@@ -358,3 +375,49 @@ func (c *bn256Pairing) Run(input []byte) ([]byte, error) {
 	}
 	return false32Byte, nil
 }
+
+type blake2F struct{}
+
+func (c *blake2F) RequiredGas(input []byte) uint64 {
+	return 1 //TODO: implement
+}
+
+func (c *blake2F) Run(input []byte) ([]byte, error) {
+	var h [8]uint64
+	h[0] = binary.BigEndian.Uint64(input[0:8])
+	h[1] = binary.BigEndian.Uint64(input[8:16])
+	h[2] = binary.BigEndian.Uint64(input[16:24])
+	h[3] = binary.BigEndian.Uint64(input[24:32])
+	h[4] = binary.BigEndian.Uint64(input[32:40])
+	h[5] = binary.BigEndian.Uint64(input[40:48])
+	h[6] = binary.BigEndian.Uint64(input[48:56])
+	h[7] = binary.BigEndian.Uint64(input[56:64])
+
+	var m [blake2.BlockSize]byte
+	copy(m[:], input[64:192])
+
+	var t [2]uint64
+	t[0] = binary.BigEndian.Uint64(input[216:224])
+	t[1] = binary.BigEndian.Uint64(input[248:256])
+
+	var f bool
+	if input[287] == 1 {
```

Comment:
Interesting -- this right here checks if the `f` is _strictly` `b 0000 0001`. So if it's `b 1111 1111`, it would be set to `false`. I think it would be more sensible to look purely at the lsb (and, naturally, to have that specified in the EIP).

Reply:
I updated the check to:
```
if input[208] == 0x00000001 {
	f = true
}
```
I am not sure if it is what you meant by looking at the least significant bit.

Reply:
I meant something like
```
if (input[208] & 0x1) != 0 {
	f = true
}
```

Reply:
I just experimented a bit with solidity. Afaict, for `boolean`, the rule seems to be `nonzero = true` (variant 1). So if we want to follow the same logic, the check would be 
```
f := (input[208] != 0)
```

Reply:
Good point, updated as requested.